.github/workflows/ci.yml

```diff
@@ -0,0 +1,50 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3.12
+      - run: python -m pip install -r dev-requirements.txt
+      - run: python -m pip install ./fluent.syntax ./fluent.runtime
+      - run: python -m flake8
+      - run: python -m mypy fluent.syntax/fluent fluent.runtime/fluent
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.7, 3.8, 3.9, "3.10", 3.11, 3.12, pypy3.9, pypy3.10]
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+      - run: python -m pip install ./fluent.syntax ./fluent.runtime
+      - run: python -m unittest discover -s fluent.syntax
+      - run: python -m unittest discover -s fluent.runtime
+  baseline:
+    runs-on: ubuntu-20.04 # https://github.com/actions/setup-python/issues/544
+    strategy:
+      matrix:
+        fluent-syntax:
+          - ./fluent.syntax
+          - fluent.syntax==0.19.0
+          - fluent.syntax==0.18.1 six
+          - fluent.syntax==0.17.0 six
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
```

Comment:
Can you use setup-python@v5 here?

Reply:
Dunno, didn't try. I intend to re-refactor this soon after landing this PR, by incrementing the minimum to around 3.9, which will let us get rid of this `ubuntu-20.04` runner as well. The action versions will get updated at that time as well.

Reply:
👍